api/src/main/java/org/apache/iceberg/encryption/EncryptingFileIO.java

```diff
@@ -109,13 +110,21 @@ public InputFile newInputFile(ManifestFile manifest) {
     }
   }
 
+  @Override
+  public InputFile newInputFile(ManifestListFile manifestList) {
+    if (manifestList.encryptedKeyMetadata() != null) {
+      ByteBuffer keyMetadata = manifestList.decryptKeyMetadata(em);
+      return newDecryptingInputFile(manifestList.location(), keyMetadata);
```

Comment:
Note that this does not have a length to pass, so this requires a few changes to the `AesGcmInputStream`.